Mathlib/Algebra/Order/Ring/Basic.lean

```diff
@@ -290,4 +290,8 @@ alias pow_two_pos_of_ne_zero := sq_pos_of_ne_zero
 lemma pow_four_le_pow_two_of_pow_two_le (h : a ^ 2 ≤ b) : a ^ 4 ≤ b ^ 2 :=
   (pow_mul a 2 2).symm ▸ pow_le_pow_left₀ (sq_nonneg a) h 2
 
+@[simp]
+lemma pow_eq_iff_eq {n : ℕ} (hn : Odd n) {a b : R} : a^n = b^n ↔ a = b :=
+  ⟨fun h ↦ (Odd.strictMono_pow hn).injective h, pow_congr _⟩
```

Comment:
This lemma should go right after `Odd.strictMono_pow`, together with (here the last one is your lemma)
```lean
lemma Odd.pow_injective {n : ℕ} (hn : Odd n) : Injective (· ^ n : R → R) :=
  hn.strictMono_pow.injective

lemma Odd.pow_lt_pow {n : ℕ} (hn : Odd n) {a b : R} : a^n < b^n ↔ a < b :=
  hn.strictMono_pow.lt_iff_lt

lemma Odd.pow_le_pow {n : ℕ} (hn : Odd n) {a b : R} : a^n ≤ b^n ↔ a ≤ b :=
  hn.strictMono_pow.le_iff_le

lemma Odd.pow_eq_pow {n : ℕ} (hn : Odd n) {a b : R} : a^n = b^n ↔ a = b :=
  hn.pow_injective.eq_iff
```
It should be in the `Odd` namespace and I don't think that it should be a `simp` lemma.